### DISCUSSION TITLE
Using Firebase SDK from multiple targets

### DISCUSSION THREAD

**Discussion by icedice** [firebase/firebase-ios-sdk#9249] [Q&A] [answered]
I'm currently in the process of splitting our monolith app into a bunch of modules (for now they are just framework targets within the same project). 

I'm implementing the Firebase SDK using carthage to build xcframeworks and adding the -objC flag for my app target.

I now need to use the SDK from other targets as well (framework modules), but this is giving me a hard time. If I link both my app target and my framework against the SDK I get a ton of those `objc[12090]: Class GDTCCTCompressionHelper is implemented in both`  If I do not link it from the framework I get stuff like `Undefined symbol: _OBJC_CLASS_$_FIRAnalytics`.

How should I approach using the SDK from multiple targets without any conflicts? Any insight would be very helpful. Thanks.

**Comment by paulb777** [comment, marked as answer]
The Firebase binary distributions are composed of static libraries. As a result they do not support linkage into a dynamic frameworks. instead the frameworks should also be statically linked. More detail at https://github.com/firebase/firebase-ios-sdk/blob/master/docs/firebase_in_libraries.md

**Reply by icedice** [reply]
Not sure how I missed that, thank I will solve it by making my modules static or injecting an abstraction into them instead.
